### PR TITLE
Brandon/reward handler fixes

### PR DESCRIPTION
**Pull request by brandonhoughton**
Fixes the reward loop in RewardForPosessingItemImplementation
Fixes double counting of craft and smelt rewards
Fixes spawning in the ground to resolve final issues with environment seeding (hopefully)

### CODE REVIEW

File: Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java [hunk 1]
```diff
@@ -69,17 +69,19 @@ public void onPickupItem(EntityItemPickupEvent event) {
...
-        if (event.player instanceof EntityPlayerMP && !event.crafting.isEmpty())
-            checkForMatch(event.crafting);
-    }
-
-    @SubscribeEvent
-    public void onItemSmelt(PlayerEvent.ItemSmeltedEvent event) {
-        if (event.player instanceof EntityPlayerMP && !event.smelting.isEmpty())
-            checkForMatch(event.smelting);
-    }
+    // These events are double counted when performed by agents
```
Comment (MadcowD) [inline Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java]: Why are these commented.

File: Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java [hunk 2]
```diff
@@ -198,49 +200,50 @@ private void removeCollectedItemCount(ItemStack is) {
...
-                    } else if (savedCollected < matcher.matchSpec.getAmount()
-                            && savedCollected + is.getCount() >= matcher.matchSpec.getAmount()) {
-                        int dimension = params.getDimension();
-                        float adjusted_reward = this.adjustAndDistributeReward(
-                                ((BlockOrItemSpecWithReward) matcher.matchSpec).getReward().floatValue(),
-                                params.getDimension(),
-                                ((BlockOrItemSpecWithReward) matcher.matchSpec).getDistribution());
-                        addCachedReward(dimension, adjusted_reward);
+                if (matcher.matches(item_stack)) {
+                    System.out.println(String.format("LOGTOPY:%s_%d.%d.%d.%d", item_stack.getUnlocalizedName(), savedCollected, maxCollected, nowCollected, matcher.matchSpec.getAmount()));
```
Comment (MadcowD) [inline Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java]: Why is there a LOG TO PY here? This is a debutg statement.

**Review by MadcowD — CHANGES_REQUESTED** [review]

File: Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java (hunk 2 above)
Comment (minerllabs) [inline Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java]: Remove logging

File: Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java (hunk 1 above)
Comment (minerllabs) [inline Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/RewardForPossessingItemImplementation.java]: Compress commented information

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Comment by minerllabs** [comment]
@MadcowD please review and merge!

**Review by MadcowD — APPROVED** [review]

**Pull request merged**